File: app/src/main/java/com/soft/credit911/ui/Notification/Adapter/NotificationDetailsAdapter.java
package com.soft.credit911.ui.Notification.Adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.soft.credit911.R;
import com.soft.credit911.datamodel.NotificationModel;
import com.soft.credit911.databinding.NotificationItemListBinding;

import java.util.List;

public class NotificationDetailsAdapter extends RecyclerView.Adapter<NotificationDetailsAdapter.NotificationViewHolder> {
    public List<NotificationModel> notificationModels;
    Context context;

    public NotificationDetailsAdapter(List<NotificationModel> notificationModels, Context context) {
        this.notificationModels = notificationModels;
        this.context = context;
    }

    @Override
    public NotificationViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        LayoutInflater layoutInflater = LayoutInflater.from(viewGroup.getContext());
        View view = layoutInflater.inflate(R.layout.notification_item_list, viewGroup, false);
        return new NotificationViewHolder(view);

    }

    @Override
    public void onBindViewHolder(NotificationViewHolder holder, int position) {
        NotificationModel notificationModelList = notificationModels.get(position);

        holder.binding.tvUserNotification.setText(notificationModelList.getUserNotification());
        holder.binding.tvStartTime.setText(notificationModelList.getStartTime());
        holder.binding.tvEndTime.setText(notificationModelList.getEndTime());
    }

    @Override
    public int getItemCount() {
        return notificationModels.size();
    }

    public class NotificationViewHolder extends RecyclerView.ViewHolder {
        public NotificationItemListBinding binding;

        public NotificationViewHolder(View itemView) {
            super(itemView);
            binding = DataBindingUtil.bind(itemView);
        }
    }
}

File: app/src/main/java/com/soft/credit911/adaptor/NotificationDetailsAdapter.kt
package com.soft.credit911.adaptor

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.soft.credit911.R
import com.soft.credit911.adaptor.NotificationDetailsAdapter.NotificationViewHolder
import com.soft.credit911.databinding.NotificationItemListBinding
import com.soft.credit911.datamodel.NotificationModel

class NotificationDetailsAdapter(
    var notificationModels: List<NotificationModel>,
    var context: Context
) : RecyclerView.Adapter<NotificationViewHolder>() {
    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): NotificationViewHolder {
        val layoutInflater = LayoutInflater.from(viewGroup.context)
        val view = layoutInflater.inflate(R.layout.notification_item_list, viewGroup, false)
        return NotificationViewHolder(view)
    }

    override fun onBindViewHolder(holder: NotificationViewHolder, position: Int) {
        val notificationModelList = notificationModels[position]
        holder.binding!!.tvUserNotification.text = notificationModelList.userNotification
        holder.binding!!.tvStartTime.text = notificationModelList.startTime
        holder.binding!!.tvEndTime.text = notificationModelList.endTime
    }

    override fun getItemCount(): Int {
        return notificationModels.size
    }

    inner class NotificationViewHolder(itemView: View?) : RecyclerView.ViewHolder(
        itemView!!
    ) {
        var binding: NotificationItemListBinding?

        init {
            binding = DataBindingUtil.bind(itemView!!)
        }
    }
}